cdap-security/src/main/java/co/cask/cdap/security/authorization/AuthorizationEnforcer.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security.authorization;
+
+import co.cask.cdap.common.UnauthorizedException;
+import co.cask.cdap.proto.id.EntityId;
+import co.cask.cdap.proto.security.Action;
+import co.cask.cdap.proto.security.Principal;
+
+/**
+ * Interface that can be used to check if the specified {@link Principal} is authorized to perform an {@link Action}
+ * on an {@link EntityId}.
+ */
+public interface AuthorizationEnforcer {
+  /**
+   * Enforces authorization for the specified {@link Principal} for the specified {@link Action} on the specified
+   * {@link EntityId}.
+   *
+   * @param entity the entity on which an action is being performed
+   * @param principal the principal that performs the actions. This could be a user, group or a role
+   * @param action the action being performed
```

Comment:
will be nice to use  links here for exception and also in place of user, group, role specially since we might remove role